# 02 - Data Observation - Données clients

Indiquez vos noms et prénoms :
-
-
-

---

<img src="https://www.smartdataforlead.com/wp-content/uploads/2018/04/37436658_m.jpg" />


Approfondissons nos compétences en matière de manipulation de données réelles.

Dans ce défi, nous explorerons les informations présentes dans une base de données clients.



> **Notre objectif est de visualiser la répartition de nos données.**



**Q1**. Téléchargez le jeu de données à l'adresse [suivante](https://drive.google.com/file/d/13EpnupLYyUBs1Y9iFjBglAvQjh4L5vru/view) puis placez-le dans le dossier 'Sample Data'.


---

**Q2**. Importez la bibliothèque `pandas` qui sera utilisée tout au long dae ce projet.

In [1]:
# Importez pandas avec l'alias pd
import pandas as pd


---

## 2. Data Exploration

Dans cette partie notre objectifs est d'explorer le jeu de données contenant les informations clients.

<img src='https://img.freepik.com/free-vector/business-documents-scanning-electronic-online-doc-with-pie-chart-infographics-data-analytics-annual-report-result-checking-man-with-magnifying-glass-concept-illustration_335657-1770.jpg?w=2000'>

Voici la liste des modules de la bibliotèque pandas qui seront utilisés dans cette partie.

- `.read_csv('chemin du fichier')` : Pour importer un jeu de données à partir de son 'nom de fichier',
- `.head(n)` : Pour afficher les n premières lignes du le jeu de données,

- `.columns` : Pour addicher la liste des variable.

- `.shape` : Pour afficher les dimensions du jeu de données (nombre de lignes et nombre de colonnes),
- `.isna().sum()` : Pour afficher le nombre de valeurs manquantes par colonnes


- `.info()` : Pour afficher les informations du jeu de données,
- `.describe()` : Pour afficher les données statistiques du jeu de données.


---


**Q3**. Importez votre le jeux de données client dans une variable nomée df. Affichez ensuite les 10 premières lignes de ce jeu de données.

In [2]:
# Importez le jeu de données dans une variable df
df = pd.read_csv('./data.csv')

# Affichez la variable df
df.head(10)

ID  Gender Ever_Married  Age Graduated     Profession  Work_Experience  \
0  462809    Male           No   22        No     Healthcare              1.0   
1  462643  Female          Yes   38       Yes       Engineer              NaN   
2  466315  Female          Yes   67       Yes       Engineer              1.0   
3  461735    Male          Yes   67       Yes         Lawyer              0.0   
4  462669  Female          Yes   40       Yes  Entertainment              NaN   
5  461319    Male          Yes   56        No         Artist              0.0   
6  460156    Male           No   32       Yes     Healthcare              1.0   
7  464347  Female           No   33       Yes     Healthcare              1.0   
8  465015  Female          Yes   61       Yes       Engineer              0.0   
9  465176  Female          Yes   55       Yes         Artist              1.0   

  Spending_Score  Family_Size  Var_1 Segmentation  
0            Low          4.0  Cat_4            D  
1        Average          3.0  Cat_4            A  
2            Low          1.0  Cat_6            B  
3           High          2.0  Cat_6            B  
4           High          6.0  Cat_6            A  
5        Average          2.0  Cat_6            C  
6            Low          3.0  Cat_6            C  
7            Low          3.0  Cat_6            D  
8            Low          3.0  Cat_7            D  
9        Average          4.0  Cat_6            C

---
 **Q4.** Affichez la liste des variables du jeu de données. 

In [3]:
#Affichez la liste des variables du jeu de données df
df.columns

Index(['ID', 'Gender', 'Ever_Married', 'Age', 'Graduated', 'Profession',
       'Work_Experience', 'Spending_Score', 'Family_Size', 'Var_1',
       'Segmentation'],
      dtype='object')

---
**Q5**. Affichez les dimensions du jeu de données ainsi que le nombre de valeurs manquantes.

In [4]:
# TODO - Affichez les dimensions
df.shape

(8068, 11)

In [5]:
# TODO - Affichez le nombre de valeurs manquantes
df.isna().sum()

ID                   0
Gender               0
Ever_Married       140
Age                  0
Graduated           78
Profession         124
Work_Experience    829
Spending_Score       0
Family_Size        335
Var_1               76
Segmentation         0
dtype: int64


---
**Q6**. Notez vos premières observations sur l'ensembles des données affichées, 
- que contient le jeu de données, quelles sont les variables, : 
- quelles sont les dimension, : Il y a 8068 personnes pour 11 colones
- quelles sont les variables qui contiennent le plus de données manquantes: La où il manque le plus de valeur manquante est le work_experience.



---
**Q7**. Affichez les informations puis les données statistics du jeu de données.


In [6]:
# TODO - Affichez les informations du jeu de données
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8068 entries, 0 to 8067
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ID               8068 non-null   int64  
 1   Gender           8068 non-null   object 
 2   Ever_Married     7928 non-null   object 
 3   Age              8068 non-null   int64  
 4   Graduated        7990 non-null   object 
 5   Profession       7944 non-null   object 
 6   Work_Experience  7239 non-null   float64
 7   Spending_Score   8068 non-null   object 
 8   Family_Size      7733 non-null   float64
 9   Var_1            7992 non-null   object 
 10  Segmentation     8068 non-null   object 
dtypes: float64(2), int64(2), object(7)
memory usage: 693.5+ KB


In [7]:
# TODO - Affichez les statistics du jeu de données
df.describe()

ID          Age  Work_Experience  Family_Size
count    8068.000000  8068.000000      7239.000000  7733.000000
mean   463479.214551    43.466906         2.641663     2.850123
std      2595.381232    16.711696         3.406763     1.531413
min    458982.000000    18.000000         0.000000     1.000000
25%    461240.750000    30.000000         0.000000     2.000000
50%    463472.500000    40.000000         1.000000     3.000000
75%    465744.250000    53.000000         4.000000     4.000000
max    467974.000000    89.000000        14.000000     9.000000


---
**Q8**. Notez vos observtions dans la zone de texte ci-dessous :
- Quelles sont les variables qualitatives ?
- Quelles sont les variables quantitatives ?
- Quel est le poids du jeu de données ?
- Que pouvez-vous dire sur l'âge des clients ?
- Que pouvez-vous dire sur la taille des familles ?
- Que pouvez-vous dire sur le nombre d'exprience professionelle ?

[Zone de texte]

## 3. Data cleaning

Dans cette partie nous allons traiter les données présentes dans notre dataframe en remplaçant les données qualitative par des données quantitatives.

<img src='https://miro.medium.com/max/1000/1*ZtO60QvYVzN3rU0VuSSU1g.jpeg'>


Ce traitement de donné nous permetra d'utiliser par la suite nos algoritmes de segmentation et de clustering.

Voici la liste des traitements que nous appliquerons à notre jeu de données :

> 1. Suppressions des lignes contenant des valeurs manquantes,

> 2. Suppressions des colonnes inutiles,

> 3. Remplacement des données quanlitatives par des données quantitatives,

> 4. Sauvegarde du jeu de données après nettoyage.


Voici la liste des modules de pandas qui seront utilisés dans cette partie :
- `df.dropna()` : Pour supprimer les lignes contenant des valeurs manquantes dans un jeu de données df. Pour concerver les modifications il faudra réassigner la variable df ==> df = df.dropna()


- `df.drop(['nom de la colone'], axis=1)` : Pour supprimer une colonne du jeu de données df. Pour concerver les modifications il faudra réassigner la variable df.


- `df['nom de la colone'].value_counts()` : Pour afficher la répartion des valeurs de la varialbe mentionnée entre crochet.


- `catégorical(df, 'nom de la colone')` : Fonction permettant de remplacer les valeurs qualitatives par des valeurs quantitatives. Pour concerver les modifications il faudra réassigner la variable df ==> df = catégorical(df, 'nom de la colone').


- `df.to_csv('fichier.csv')` : Pour sauvegarder votre jeu de données.


---
**Q9** Exécuter la cellule suivante, elle contient la fonction `catégorical` permettant le traitement de nos données qualitatives.

In [8]:
def catégorical(df, column):
    df = df.dropna()
    liste_ = list(df[column].value_counts().index)
    df[column] = df[column].apply(lambda x: liste_.index(x))
    return df

---
**Q10**. Supprimez les lignes contenant des données manquantes puis affichez le jeu de données. Notez ensuite vos remarques dans une zone de texte.

In [9]:
#TODO -  Supprimez les données manquantes
df = df.dropna()
df

ID  Gender Ever_Married  Age Graduated  Profession  Work_Experience  \
0     462809    Male           No   22        No  Healthcare              1.0   
2     466315  Female          Yes   67       Yes    Engineer              1.0   
3     461735    Male          Yes   67       Yes      Lawyer              0.0   
5     461319    Male          Yes   56        No      Artist              0.0   
6     460156    Male           No   32       Yes  Healthcare              1.0   
...      ...     ...          ...  ...       ...         ...              ...   
8062  463002    Male          Yes   41       Yes      Artist              0.0   
8064  464685    Male           No   35        No   Executive              3.0   
8065  465406  Female           No   33       Yes  Healthcare              1.0   
8066  467299  Female           No   27       Yes  Healthcare              1.0   
8067  461879    Male          Yes   37       Yes   Executive              0.0   

     Spending_Score  Family_Size  Var_1 Segmentation  
0               Low          4.0  Cat_4            D  
2               Low          1.0  Cat_6            B  
3              High          2.0  Cat_6            B  
5           Average          2.0  Cat_6            C  
6               Low          3.0  Cat_6            C  
...             ...          ...    ...          ...  
8062           High          5.0  Cat_6            B  
8064            Low          4.0  Cat_4            D  
8065            Low          1.0  Cat_6            D  
8066            Low          4.0  Cat_6            B  
8067        Average          3.0  Cat_4            B  

[6665 rows x 11 columns]

[Zone de texte]

---
**Q11**. Supprimez la colonne 'Var_1' ainsi que la colonne 'ID' du jeu de données.


In [10]:
# Suppression de la variable Var_1
df = df.drop('Var_1', axis=1)
df

ID  Gender Ever_Married  Age Graduated  Profession  Work_Experience  \
0     462809    Male           No   22        No  Healthcare              1.0   
2     466315  Female          Yes   67       Yes    Engineer              1.0   
3     461735    Male          Yes   67       Yes      Lawyer              0.0   
5     461319    Male          Yes   56        No      Artist              0.0   
6     460156    Male           No   32       Yes  Healthcare              1.0   
...      ...     ...          ...  ...       ...         ...              ...   
8062  463002    Male          Yes   41       Yes      Artist              0.0   
8064  464685    Male           No   35        No   Executive              3.0   
8065  465406  Female           No   33       Yes  Healthcare              1.0   
8066  467299  Female           No   27       Yes  Healthcare              1.0   
8067  461879    Male          Yes   37       Yes   Executive              0.0   

     Spending_Score  Family_Size Segmentation  
0               Low          4.0            D  
2               Low          1.0            B  
3              High          2.0            B  
5           Average          2.0            C  
6               Low          3.0            C  
...             ...          ...          ...  
8062           High          5.0            B  
8064            Low          4.0            D  
8065            Low          1.0            D  
8066            Low          4.0            B  
8067        Average          3.0            B  

[6665 rows x 10 columns]

In [11]:
# Suppression de la variable ID
df = df.drop('ID', axis=1)
df

Gender Ever_Married  Age Graduated  Profession  Work_Experience  \
0       Male           No   22        No  Healthcare              1.0   
2     Female          Yes   67       Yes    Engineer              1.0   
3       Male          Yes   67       Yes      Lawyer              0.0   
5       Male          Yes   56        No      Artist              0.0   
6       Male           No   32       Yes  Healthcare              1.0   
...      ...          ...  ...       ...         ...              ...   
8062    Male          Yes   41       Yes      Artist              0.0   
8064    Male           No   35        No   Executive              3.0   
8065  Female           No   33       Yes  Healthcare              1.0   
8066  Female           No   27       Yes  Healthcare              1.0   
8067    Male          Yes   37       Yes   Executive              0.0   

     Spending_Score  Family_Size Segmentation  
0               Low          4.0            D  
2               Low          1.0            B  
3              High          2.0            B  
5           Average          2.0            C  
6               Low          3.0            C  
...             ...          ...          ...  
8062           High          5.0            B  
8064            Low          4.0            D  
8065            Low          1.0            D  
8066            Low          4.0            B  
8067        Average          3.0            B  

[6665 rows x 9 columns]

---
**Q12**. Traitement de la colonne `Gender`

- Affichez la répartition des valeurs de la colonne `Gender`,
- Notez vos observations dans la zonne de texte,
- Appliquez la fonction `catégorial` à la colonne `Gender`,

In [12]:
# Affichez la répartition de la colonne 'Gender'
df['Gender']

0         Male
2       Female
3         Male
5         Male
6         Male
         ...  
8062      Male
8064      Male
8065    Female
8066    Female
8067      Male
Name: Gender, Length: 6665, dtype: object

[Zone de texte]

In [13]:
# Remplacez les valeurs qualitatives par des valeurs quantitatives
df = catégorical(df, 'Gender')
df

Gender Ever_Married  Age Graduated  Profession  Work_Experience  \
0          0           No   22        No  Healthcare              1.0   
2          1          Yes   67       Yes    Engineer              1.0   
3          0          Yes   67       Yes      Lawyer              0.0   
5          0          Yes   56        No      Artist              0.0   
6          0           No   32       Yes  Healthcare              1.0   
...      ...          ...  ...       ...         ...              ...   
8062       0          Yes   41       Yes      Artist              0.0   
8064       0           No   35        No   Executive              3.0   
8065       1           No   33       Yes  Healthcare              1.0   
8066       1           No   27       Yes  Healthcare              1.0   
8067       0          Yes   37       Yes   Executive              0.0   

     Spending_Score  Family_Size Segmentation  
0               Low          4.0            D  
2               Low          1.0            B  
3              High          2.0            B  
5           Average          2.0            C  
6               Low          3.0            C  
...             ...          ...          ...  
8062           High          5.0            B  
8064            Low          4.0            D  
8065            Low          1.0            D  
8066            Low          4.0            B  
8067        Average          3.0            B  

[6665 rows x 9 columns]

---
**Q13**. Traitement de la colonne `Ever_Married`

- Affichez la répartition des valeurs de la colonne `Ever_Married`,
- Notez vos observations dans la zonne de texte,
- Appliquez la fonction `catégorial` à la colonne `Ever_Married`,

In [14]:
# Affichez la répartition de la colonne 'Ever_Married'
df['Ever_Married']

0        No
2       Yes
3       Yes
5       Yes
6        No
       ... 
8062    Yes
8064     No
8065     No
8066     No
8067    Yes
Name: Ever_Married, Length: 6665, dtype: object

[Zone de texte]

In [15]:
# Remplacez les valeurs qualitatives par des valeurs quantitatives
df = catégorical(df, 'Ever_Married')
df

Gender  Ever_Married  Age Graduated  Profession  Work_Experience  \
0          0             1   22        No  Healthcare              1.0   
2          1             0   67       Yes    Engineer              1.0   
3          0             0   67       Yes      Lawyer              0.0   
5          0             0   56        No      Artist              0.0   
6          0             1   32       Yes  Healthcare              1.0   
...      ...           ...  ...       ...         ...              ...   
8062       0             0   41       Yes      Artist              0.0   
8064       0             1   35        No   Executive              3.0   
8065       1             1   33       Yes  Healthcare              1.0   
8066       1             1   27       Yes  Healthcare              1.0   
8067       0             0   37       Yes   Executive              0.0   

     Spending_Score  Family_Size Segmentation  
0               Low          4.0            D  
2               Low          1.0            B  
3              High          2.0            B  
5           Average          2.0            C  
6               Low          3.0            C  
...             ...          ...          ...  
8062           High          5.0            B  
8064            Low          4.0            D  
8065            Low          1.0            D  
8066            Low          4.0            B  
8067        Average          3.0            B  

[6665 rows x 9 columns]

---
**Q14**. Traitement de la colonne `Graduated`

- Affichez la répartition des valeurs de la colonne `Graduated`,
- Notez vos observations dans la zonne de texte,
- Appliquez la fonction `catégorial` à la colonne `Graduated`,

In [16]:
# Affichez la répartition de la colonne 'Graduated'
df['Graduated']

0        No
2       Yes
3       Yes
5        No
6       Yes
       ... 
8062    Yes
8064     No
8065    Yes
8066    Yes
8067    Yes
Name: Graduated, Length: 6665, dtype: object

[Zone de texte]

In [17]:
# Remplacez les valeurs qualitatives par des valeurs quantitatives
df = catégorical(df, 'Graduated')
df

Gender  Ever_Married  Age  Graduated  Profession  Work_Experience  \
0          0             1   22          1  Healthcare              1.0   
2          1             0   67          0    Engineer              1.0   
3          0             0   67          0      Lawyer              0.0   
5          0             0   56          1      Artist              0.0   
6          0             1   32          0  Healthcare              1.0   
...      ...           ...  ...        ...         ...              ...   
8062       0             0   41          0      Artist              0.0   
8064       0             1   35          1   Executive              3.0   
8065       1             1   33          0  Healthcare              1.0   
8066       1             1   27          0  Healthcare              1.0   
8067       0             0   37          0   Executive              0.0   

     Spending_Score  Family_Size Segmentation  
0               Low          4.0            D  
2               Low          1.0            B  
3              High          2.0            B  
5           Average          2.0            C  
6               Low          3.0            C  
...             ...          ...          ...  
8062           High          5.0            B  
8064            Low          4.0            D  
8065            Low          1.0            D  
8066            Low          4.0            B  
8067        Average          3.0            B  

[6665 rows x 9 columns]

---
**Q15**. Traitement de la colonne `Profession`

- Affichez la répartition des valeurs de la colonne `Profession`,
- Notez vos observations dans la zonne de texte,
- Appliquez la fonction `catégorial` à la colonne `Profession`,

In [18]:
#Affichez la répartition de la colonne 'Profession'
df['Profession']

0       Healthcare
2         Engineer
3           Lawyer
5           Artist
6       Healthcare
           ...    
8062        Artist
8064     Executive
8065    Healthcare
8066    Healthcare
8067     Executive
Name: Profession, Length: 6665, dtype: object

[Zone de texte]

In [19]:
# Remplacez les valeurs qualitatives par des valeurs quantitatives
df = catégorical(df, 'Profession')
df

Gender  Ever_Married  Age  Graduated  Profession  Work_Experience  \
0          0             1   22          1           1              1.0   
2          1             0   67          0           4              1.0   
3          0             0   67          0           6              0.0   
5          0             0   56          1           0              0.0   
6          0             1   32          0           1              1.0   
...      ...           ...  ...        ...         ...              ...   
8062       0             0   41          0           0              0.0   
8064       0             1   35          1           5              3.0   
8065       1             1   33          0           1              1.0   
8066       1             1   27          0           1              1.0   
8067       0             0   37          0           5              0.0   

     Spending_Score  Family_Size Segmentation  
0               Low          4.0            D  
2               Low          1.0            B  
3              High          2.0            B  
5           Average          2.0            C  
6               Low          3.0            C  
...             ...          ...          ...  
8062           High          5.0            B  
8064            Low          4.0            D  
8065            Low          1.0            D  
8066            Low          4.0            B  
8067        Average          3.0            B  

[6665 rows x 9 columns]

---
**Q16**. Traitement de la colonne `Spending_Score`

- Affichez la répartition des valeurs de la colonne `Spending_Score`,
- Notez vos observations dans la zonne de texte,
- Appliquez la fonction `catégorial` à la colonne `Spending_Score`,

In [20]:
# Affichez la répartition de la colonne 'Spending_Score'
df['Spending_Score']

0           Low
2           Low
3          High
5       Average
6           Low
         ...   
8062       High
8064        Low
8065        Low
8066        Low
8067    Average
Name: Spending_Score, Length: 6665, dtype: object

[Zone de texte]

In [21]:
# Remplacez les valeurs qualitatives par des valeurs quantitatives
df = catégorical(df, 'Spending_Score')
df

Gender  Ever_Married  Age  Graduated  Profession  Work_Experience  \
0          0             1   22          1           1              1.0   
2          1             0   67          0           4              1.0   
3          0             0   67          0           6              0.0   
5          0             0   56          1           0              0.0   
6          0             1   32          0           1              1.0   
...      ...           ...  ...        ...         ...              ...   
8062       0             0   41          0           0              0.0   
8064       0             1   35          1           5              3.0   
8065       1             1   33          0           1              1.0   
8066       1             1   27          0           1              1.0   
8067       0             0   37          0           5              0.0   

      Spending_Score  Family_Size Segmentation  
0                  0          4.0            D  
2                  0          1.0            B  
3                  2          2.0            B  
5                  1          2.0            C  
6                  0          3.0            C  
...              ...          ...          ...  
8062               2          5.0            B  
8064               0          4.0            D  
8065               0          1.0            D  
8066               0          4.0            B  
8067               1          3.0            B  

[6665 rows x 9 columns]

---
**Q17**. Traitement de la colonne `Segmentation`

- Affichez la répartition des valeurs de la colonne `Segmentation`,
- Notez vos observations dans la zonne de texte,
- Appliquez la fonction `catégorial` à la colonne `Segmentation`,

In [22]:
#TODO - Affichez la répartition de la colonne 'Segmentation'
df['Segmentation']

0       D
2       B
3       B
5       C
6       C
       ..
8062    B
8064    D
8065    D
8066    B
8067    B
Name: Segmentation, Length: 6665, dtype: object

[Zone de texte]

In [23]:
# Remplacez les valeurs qualitatives par des valeurs quantitatives
df = catégorical(df, 'Segmentation')
df

Gender  Ever_Married  Age  Graduated  Profession  Work_Experience  \
0          0             1   22          1           1              1.0   
2          1             0   67          0           4              1.0   
3          0             0   67          0           6              0.0   
5          0             0   56          1           0              0.0   
6          0             1   32          0           1              1.0   
...      ...           ...  ...        ...         ...              ...   
8062       0             0   41          0           0              0.0   
8064       0             1   35          1           5              3.0   
8065       1             1   33          0           1              1.0   
8066       1             1   27          0           1              1.0   
8067       0             0   37          0           5              0.0   

      Spending_Score  Family_Size  Segmentation  
0                  0          4.0             0  
2                  0          1.0             3  
3                  2          2.0             3  
5                  1          2.0             1  
6                  0          3.0             1  
...              ...          ...           ...  
8062               2          5.0             3  
8064               0          4.0             0  
8065               0          1.0             0  
8066               0          4.0             3  
8067               1          3.0             3  

[6665 rows x 9 columns]

---
**Q18**. Sauvegardez votre jeu de données dans un fichier nommé 'data_clean.csv'

In [24]:
# Sauvegardez votre jeu de données


# 4. Importez un jeu de données quelquonque à partir du des banques de données suivantes :    


- https://www.data.gouv.fr/fr/
- https://data.education.gouv.fr/pages/accueil/
- https://opendata.paris.fr/pages/home/
- https://data.ratp.fr/explore/?sort=modified
- https://opendata.edf.fr/pages/welcome/?flg=fr 
- https://msropendata.com/
- https://www.kaggle.com/
- https://www.reddit.com/r/datasets/

Affichez par la suite : 
- la dimension du jeu de données,
- les données statistiques,
- le détail des variables,
- et procédez au traitement d'une variable qualitative.

In [25]:
d = pd.read_csv('./fr-en-calendrier-scolaire.csv')
d

Description   Population              Date de début  \
0        Vacances de Printemps            -  2010-04-26T00:00:00+02:00   
1               Vacances d'Été  Enseignants  2010-07-08T00:00:00+02:00   
2     Vacances de la Toussaint            -  2010-10-25T00:00:00+02:00   
3        Vacances de Printemps            -  2012-04-23T00:00:00+02:00   
4               Vacances d'Été       Élèves  2012-07-09T00:00:00+02:00   
...                        ...          ...                        ...   
1895  Début des Vacances d'Été            -  2026-07-04T00:00:00+02:00   
1896  Début des Vacances d'Été            -  2026-07-04T00:00:00+02:00   
1897  Début des Vacances d'Été            -  2026-07-04T00:00:00+02:00   
1898  Début des Vacances d'Été            -  2026-07-04T00:00:00+02:00   
1899  Début des Vacances d'Été            -  2026-07-04T00:00:00+02:00   

                    Date de fin         Académies   Zones annee_scolaire  
0     2010-05-10T00:00:00+02:00             Corse   Corse      2009-2010  
1     2010-09-08T00:00:00+02:00             Corse   Corse      2009-2010  
2     2010-11-02T00:00:00+01:00             Corse   Corse      2010-2011  
3     2012-05-07T00:00:00+02:00             Corse   Corse      2011-2012  
4     2012-09-10T00:00:00+02:00             Corse   Corse      2011-2012  
...                         ...               ...     ...            ...  
1895  2026-07-04T00:00:00+02:00  Clermont-Ferrand  Zone A      2025-2026  
1896  2026-07-04T00:00:00+02:00              Lyon  Zone A      2025-2026  
1897  2026-07-04T00:00:00+02:00              Nice  Zone B      2025-2026  
1898  2026-07-04T00:00:00+02:00     Orléans-Tours  Zone B      2025-2026  
1899  2026-07-04T00:00:00+02:00       Montpellier  Zone C      2025-2026  

[1900 rows x 7 columns]

In [26]:
d.shape

(1900, 7)

In [27]:
d.describe()

Description Population              Date de début  \
count             1900       1900                       1900   
unique              30         19                        287   
top     Vacances d'Été          -  2019-07-06T00:00:00+02:00   
freq               458       1415                         60   

                      Date de fin Académies   Zones annee_scolaire  
count                        1900      1900    1900           1900  
unique                        330        36      13             22  
top     2019-01-07T00:00:00+01:00     Corse  Zone B      2021-2022  
freq                           31        85     675            232

In [28]:
d.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1900 entries, 0 to 1899
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Description     1900 non-null   object
 1   Population      1900 non-null   object
 2   Date de début   1900 non-null   object
 3   Date de fin     1900 non-null   object
 4   Académies       1900 non-null   object
 5   Zones           1900 non-null   object
 6   annee_scolaire  1900 non-null   object
dtypes: object(7)
memory usage: 104.0+ KB


In [29]:
d.isna().sum()

Description       0
Population        0
Date de début     0
Date de fin       0
Académies         0
Zones             0
annee_scolaire    0
dtype: int64